dev/com.ibm.ws.config/src/com/ibm/ws/config/xml/internal/ConfigRefresher.java

```diff
@@ -136,7 +157,12 @@ private synchronized void doRefresh(Map<String, DeltaType> variableDelta) {
             // Let the notification show that we got an error while making the configuration changes
             configUpdatesDelivered.setResult(e);
         } finally {
-            changesEnded(configUpdatesDelivered);
+            if (CheckpointPhase.getPhase().restored() && changesEndedHook.runMultiThreaded) {
+                // The hook will invoke changesEnded(configUpdatesDelivered)
+                changesEndedHook.configUpdatesDelivered = configUpdatesDelivered;
```

Comment:
This could be a non-issue ...

It is not clear to me if multiple `RuntimeUpdateNotification` could be produced during a restore.  It would involve multiple calls to `com.ibm.ws.config.xml.internal.ConfigRefresher.doRefresh(Map<String, DeltaType>)`.  If that is possible then the "last one wins" and the earlier notifications would not get fired.

I question if `changesEndedHook.configUpdatesDelivered` should be some thread-safe queue that is drained in the `restore()` method.

Reply:
I've uploaded the requested changes to [Change hook registration and queue config update notifications](https://github.com/OpenLiberty/open-liberty/pull/26508/commits/85de2e5cd7a82f44e204747c57657bb931c98e8b).